src/parsing/globalModuleCache.ts

```diff
@@ -57,3 +70,27 @@ export const enhanceStaticContextWithModule = (staticContext: StaticContext, uri
 		);
 	});
 };
+
+/**
+ * Perform static compilation on all registered modules
+ *
+ * This also happens when the first XPath is executed, but it is good practice to run this function after all XQuery modules are registered.
+ *
+ * @see registerXQueryModule
+ *
+ * @public
+ */
+export const performStaticCompilationOnModules = () => {
+	Object.keys(loadedModulesByNamespaceURI).forEach((namespaceURI) => {
+		const moduleContents = loadedModulesByNamespaceURI[namespaceURI];
+		if (moduleContents.performStaticAnalysis) {
+			try {
+				moduleContents.performStaticAnalysis(moduleContents);
+			} catch (error: unknown) {
+				moduleContents.performStaticAnalysis = null;
```

Comment:
What would happen the next time someone tries to call this module? Is it left in a consistent state, or should we put a separate try/catch around each individual registered module's performStaticAnalysis call to make sure valid modules do not skip static analysis if an earlier module in the same namespace fails?

Reply:
It throws and then set up for future compilation. A static error may be resolved later